File: EaterSpring/src/main/kotlin/xyz/purema/eater/controllers/CustomerController.kt
package xyz.purema.eater.controllers

import org.springframework.web.bind.annotation.*
import xyz.purema.eater.dtos.form.LoginForm
import xyz.purema.eater.dtos.form.RegisterForm
import xyz.purema.eater.services.CustomerService
import javax.validation.Valid

@RestController
@RequestMapping("/v1/customer")
class CustomerController(
    private val customerService: CustomerService
) {
    @PostMapping("/register")
    fun register(@Valid @RequestBody form: RegisterForm) = customerService.register(form)

    @PostMapping("/login")
    fun login(@Valid @RequestBody form: LoginForm) = customerService.login(form)

    @GetMapping
    fun getProfile() = customerService.getProfile()
}

File: EaterNet/Eater/Controllers/CustomerController.cs
using Eater.Dtos;
using Eater.Dtos.Form;
using Eater.Interfaces;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace Eater.Controllers
{
    [Route("v1/customer")]
    [ApiController]
    [Authorize]
    public class CustomerController : ControllerBase
    {
        private readonly ICustomerService _customerService;

        public CustomerController(ICustomerService customerService)
        {
            _customerService = customerService;
        }

        [HttpPost("register")]
        [AllowAnonymous]
        public async Task<AuthResultDto> Register([FromBody] RegisterForm form) => await _customerService.Register(form);

        [HttpPost("login")]
        [AllowAnonymous]
        public async Task<AuthResultDto> Login([FromBody] LoginForm form) => await _customerService.Login(form);

        [HttpGet]
        public async Task<CustomerDto> GetProfile() => await _customerService.GetProfile();
    }
}
